django/db/models/query_utils.py

```diff
@@ -192,20 +193,61 @@ def _check_parent_chain(self, instance):
             return getattr(instance, link_field.attname)
         return None
 
+    # proxy methods to redirect to field methods to simplify path for users
+    def set_up_proxy_methods(self):
+        self.register_lookup = self.field.register_lookup
+        self._unregister_lookup = self.field._unregister_lookup
+        self.get_lookups = self.field.get_lookups
+        self.get_lookup = self.field.get_lookup
+        self.get_transform = self.field.get_transform
+
+
+class classorinstancemethod:
+    """
+    A class used in RegisterLookupMixin which returns the necessary partial
+    functions depending on whether it was called by a Field instance or a Field
+    class
+    """
+
+    def __init__(self, class_method, instance_method):
+        self.class_method = class_method
+        self.instance_method = instance_method
+
+    def __get__(self, instance, owner):
+        if instance is None:
+            partial_class_method = functools.partial(self.class_method, owner)
+            return self.get_partial_method(partial_class_method, self.class_method)
+        partial_instance_method = functools.partial(self.instance_method, instance)
+        return self.get_partial_method(partial_instance_method, self.class_method)
+
+    @staticmethod
+    def get_partial_method(partial_function, class_method):
+        method_name = partial_function.func.__name__
+        if method_name in ["get_class_lookups", "get_instance_lookups"]:
+            partial_function.cache_clear = class_method.cache_clear
```

Comment:
This seems a bit odd to have `classorinstancemethod` specialize these two here.

Reply:
`get_class_lookups` caches lookup using `@functools.lru_cache` decorator. So I wanted `get_instance_lookups` to also have the functionality to `clear_cache`. But maybe this would be necessary as I think about it, since its only going to be used by Django developers. Maybe I'll change the test that fails
![image](https://user-images.githubusercontent.com/90314608/181056877-4e6cc801-b5f8-4e1d-ab80-dff1c25e5d47.png)
Replacing `get_lookups` by `get_class_lookups` would solve this.

Reply:
Done👍